examples/plot_casino_example.py

```diff
@@ -0,0 +1,149 @@
+# and generate a series of rolls that someone at the casino would
+# observe.
+
+# make our generative model with two components, a fair die and a
+# loaded die
+gen_model = hmm.MultinomialHMM(n_components=2, random_state=99)
+
+# the first state is the fair die so let's start there so no one
+# catches on rigth away
+gen_model.startprob_ = np.array([1.0, 0.0])
+
+# now let's say that we sneak the loaded die in:
+# here, we have a 95% chance to continue using the fair die and a 5%
+# chance to switch to the loaded die
+# when we enter the loaded die state, we have a 90% chance of staying
+# in that state and a 10% chance of leaving
+gen_model.transmat_ = np.array([[0.95, 0.05],
+                                [0.1, 0.9]])
+
+# now let's set the emission means:
+# the first state is a fair die with equal probabilities and the
+# second is loaded by being biased toward rolling a six
+gen_model.emissionprob_ = \
+    np.array([[1 / 6, 1 / 6, 1 / 6, 1 / 6, 1 / 6, 1 / 6],
+              [1 / 10, 1 / 10, 1 / 10, 1 / 10, 1 / 10, 1 / 2]])
+
+# simulate the loaded dice rolls
+rolls, gen_states = gen_model.sample(30000)
+
+# plot states over time, let's just look at the first rolls for clarity
+fig, ax = plt.subplots()
+ax.plot(gen_states[:500])
+ax.set_title('States over time')
+ax.set_xlabel('Time (# of rolls)')
+ax.set_ylabel('State')
+fig.show()
+
+# plot rolls for the fair and loaded states
+fig, ax = plt.subplots()
+ax.hist(rolls[gen_states == 0], label='fair', alpha=0.5,
+        bins=np.arange(7) - 0.5, density=True)
+ax.hist(rolls[gen_states == 1], label='loaded', alpha=0.5,
+        bins=np.arange(7) - 0.5, density=True)
+ax.set_title('Roll probabilities by state')
+ax.set_xlabel('Count')
+ax.set_ylabel('Roll')
+ax.legend()
+fig.show()
+
+# %%
+# Now, let's see if we can recover our hidden states, transmission matrix
+# and emission probabilities.
+
+# split our data into training and validation sets (50/50 split)
+X_train = rolls[:rolls.shape[0] // 2]
+X_validate = rolls[rolls.shape[0] // 2:]
+
+# check optimal score
+gen_score = gen_model.score(X_validate)
+
+best_score = best_model = None
+n_fits = 50
+# we know transitions are rare (otherwise the casino would get caught!)
+# so let's have an uniform random prior distribution on our transitions
+# between 0 and 0.25 train our model faster (than uniform random 0 to 1)
+rng = np.random.default_rng(seed=13)  # seed random number generator
+fair_to_loaded_priors = rng.random(size=n_fits) / 4
+loaded_to_fair_priors = rng.random(size=n_fits) / 4
+for idx in range(n_fits):
+    model = hmm.MultinomialHMM(
+        n_components=2, random_state=idx,
+        init_params='se')  # don't init transition, set it below
+    # we need to initialize with random transition matrix probabilities
+    # because the default is an even likelihood transition
+    fair_to_loaded = fair_to_loaded_priors[idx]
+    loaded_to_fair = loaded_to_fair_priors[idx]
```

Comment:
I would just draw the random init here (`fair_to_loaded = 0.25*random()`); there's not advantage of drawing them all at the beginning and indexing.

Also hmmlearn currently doesn't require the "new" numpy random api and I think we should keep using that in the example as well (to avoid having a dependency on numpy 1.17).

Reply:
Ok just using `np.random.random`